commons/tool/src/webapp/js/commons_utils.js

```diff
@@ -419,168 +433,260 @@ commons.utils = {
+            const template = Handlebars.templates['posts_placeholders'];
+            $('#commons-posts').append(template({ posts }, {helpers: commonsHelpers}));
 
-                commons.currentPosts = commons.currentPosts.concat(posts);
+            // Now render them into their placeholders
+            posts.forEach(p => commons.utils.renderPost(p, `commons-post-${p.id}`));
+            
+            // Apply user likes once for all posts
+            commons.utils.getUserLikes();
 
-                if (commons.page == 0 && data.postsTotal > 0) {
-                    $('#commons-body-toggle').show();
+            try {
+                if (window.frameElement) {
+                    setMainFrameHeight(window.frameElement.id);
                 }
-
-                commons.postsRendered += posts.length;
-
-                commons.utils.addFormattedDatesToPosts(posts);
-
-                // Add the next batch of placeholders to the post list
-                var t = Handlebars.templates['posts_placeholders'];
-                $('#commons-posts').append(t({ posts: posts }, {helpers: commonsHelpers}));
-
-                $(document).ready(function () {
-
-                    // Now render them into their placeholders
-                    posts.forEach(function (p) { commons.utils.renderPost(p, 'commons-post-' + p.id); });
-
-                    loadImage.hide();
-                    try {
-                        if (window.frameElement) {
-                            setMainFrameHeight(window.frameElement.id);
-                        }
-                    } catch (err) {
-                        // This is likely under an LTI provision scenario.
-                        // XSS protection will block this call.
-                    }
+            } catch (err) {
+                // This is likely under an LTI provision scenario.
+                // XSS protection will block this call.
+            }
+            
+            commons.page += 1;
+            console.debug('Incremented page counter to', commons.page);
+
+            // Add a "Load More" button if this is not the first page and we have more posts to load
+            if (commons.page > 0 && data.status !== 'END') {
+                // Remove any existing load more button
+                $('#commons-load-more').remove();
+                
+                // Add a new load more button
+                $('#commons-posts').append(
+                    '<div id="commons-load-more" class="commons-load-more">' +
+                    '<button class="btn btn-primary">' + commons.i18n['load_more'] + '</button>' +
+                    '</div>'
+                );
+                
+                // Add click handler for the load more button
+                $('#commons-load-more button').click(function() {
+                    console.debug('Load more button clicked');
+                    $(this).prop('disabled', true).text(commons.i18n['loading_posts']);
```

Comment:
```suggestion
                    $(this).prop('disabled', true).text(commons.i18n.loading_posts);
```

Reply:
ok i will change all in the file in one go .....